crates/dbsp/src/operator/input.rs

```diff
@@ -338,15 +327,17 @@ impl RootCircuit {
     /// operator:
     ///
     /// ```text
-    /// time │      input commands               │content of the        │ stream returned by         │  comment
-    ///      │                                   │input map             │ `add_input_map`            │
-    /// ─────┼───────────────────────────────────┼──────────────────────┼────────────────────────────┼───────────────────────────────────────────────────────
-    ///    1 │{(1,Some("foo"), (2,Some("bar"))}  │{(1,"foo"),(2,"bar")} │ {(1,"foo",+1),(2,"bar",+1)}│
-    ///    2 │{(1,Some("foo"), (2,Some("baz"))}  │{(1,"foo"),(2,"baz")} │ {(2,"bar",-1),(2,"baz",+1)}│ Ignore duplicate insert of (1,"foo"). New value
-    ///      |                                   |                      |                            | "baz" for key 2 overwrites the old value "bar".
-    ///    3 │{(1,None),(2,Some("bar")),(2,None)}│{}                    │ {(1,"foo",-1),(2,"baz",-1)}│ Delete both keys. Upsert (2,"bar") is overridden
-    ///      |                                   |                      |                            | by subsequent delete command.
-    /// ─────┴───────────────────────────────────┴──────────────────────┴────────────────────────────┴────────────────────────────────────────────────────────
+    /// time │      input commands                     │content of the        │ stream returned by         │  comment
+    ///      │                                         │input map             │ `add_input_map`            │
+    /// ─────┼─────────────────────────────────────────┼──────────────────────┼────────────────────────────┼───────────────────────────────────────────────────────
+    ///    1 │{(1,Insert("foo"), (2,Insert("bar"))}    │{(1,"foo"),(2,"bar")} │ {(1,"foo",+1),(2,"bar",+1)}│
+    ///    2 │{(1,Insert("foo"), (2,Insert("baz"))}    │{(1,"foo"),(2,"baz")} │ {(2,"bar",-1),(2,"baz",+1)}│ Ignore duplicate insert of (1,"foo"). New value
+    ///      |                                         |                      |                            | "baz" for key 2 overwrites the old value "bar".
+    ///    3 │{(1,Delete),(2,Insert("bar")),(2,Delete)}│{}                    │ {(1,"foo",-1),(2,"baz",-1)}│ Delete both keys. Upsert (2,"bar") is overridden
+    ///      |                                         |                      |                            | by subsequent delete command.
+    ///    4 |{(1,Update("new")), (2,Update("bar"))}   |{(1,"foo")}           | {(1,"new")}                | Note that the second update is ignored because
```

Comment:
When we have an error stream an update to a missing key could cause a warning.

Reply:
yep, there are extensive comments to that effect in the implementation of the `input_upsert` operator, but right now it just gets ignored.